### PR TITLE
add reStartRecordingLive function

### PR DESCRIPTION
**Pull request by fengcunhan**
You can continues record after pause recording

### CODE REVIEW

File: Sources/WhisperKit/Core/AudioProcessor.swift [hunk 1]
```diff
@@ -587,6 +595,20 @@ public extension AudioProcessor {
...
+    
+    func reStartRecordingLive(inputDeviceID: DeviceID? = nil, callback: (([Float]) -> Void)? = nil) throws {
+        try? setupAudioSessionForDevice()
+        
+        if inputDeviceID == lastInputDevice{
+            try audioEngine?.start()
+        }else{
+            audioEngine = try setupEngine(inputDeviceID: inputDeviceID)
+        }
+        // Set the callback
```
Comment (ZachNagengast) [inline Sources/WhisperKit/Core/AudioProcessor.swift:610]:
```suggestion

        // Set the callback
```

File: Sources/WhisperKit/Core/AudioProcessor.swift [hunk 2]
```diff
@@ -587,6 +595,20 @@ public extension AudioProcessor {
 
         // Set the callback
         audioBufferCallback = callback
+        
+        lastInputDevice = inputDeviceID
+    }
+    
+    func reStartRecordingLive(inputDeviceID: DeviceID? = nil, callback: (([Float]) -> Void)? = nil) throws {
```
Comment (ZachNagengast) [inline Sources/WhisperKit/Core/AudioProcessor.swift:602]:
```suggestion
    func resumeRecordingLive(inputDeviceID: DeviceID? = nil, callback: (([Float]) -> Void)? = nil) throws {
```

File: Sources/WhisperKit/Core/AudioProcessor.swift [hunk 3]
```diff
@@ -64,13 +64,20 @@ public protocol AudioProcessing {
...
 
 /// Overrideable default methods for AudioProcessing
 public extension AudioProcessing {
     func startRecordingLive(inputDeviceID: DeviceID? = nil, callback: (([Float]) -> Void)?) throws {
         try startRecordingLive(inputDeviceID: inputDeviceID, callback: callback)
     }
+    
+    func reStartRecordingLive(inputDeviceID: DeviceID? = nil, callback: (([Float]) -> Void)?) throws {
+        try reStartRecordingLive(inputDeviceID: inputDeviceID, callback: callback)
+    }
```
Comment (ZachNagengast) [inline Sources/WhisperKit/Core/AudioProcessor.swift:80]:
```suggestion
    func resumeRecordingLive(inputDeviceID: DeviceID? = nil, callback: (([Float]) -> Void)?) throws {
        try resumeRecordingLive(inputDeviceID: inputDeviceID, callback: callback)
    }
```

File: Sources/WhisperKit/Core/AudioProcessor.swift [hunk 4]
```diff
@@ -64,13 +64,20 @@ public protocol AudioProcessing {
 
     /// Stops recording and cleans up resources
     func stopRecording()
+    
+    /// Restart recording audio from the specified input device,you can continius record after puase record
+    func reStartRecordingLive(inputDeviceID: DeviceID?, callback: (([Float]) -> Void)?) throws
```
Comment (ZachNagengast) [inline Sources/WhisperKit/Core/AudioProcessor.swift:69]:
```suggestion
    /// Resume recording audio from the specified input device, appending to continuous `audioArray` after pause
    func resumeRecordingLive(inputDeviceID: DeviceID?, callback: (([Float]) -> Void)?) throws
```

**Review by ZachNagengast — CHANGES_REQUESTED** [review]
This is great, seems like there was the pause feature but was missing resume, suggested a couple changes for clarity.

File: Sources/WhisperKit/Core/AudioProcessor.swift [hunk 5]
```diff
@@ -587,6 +595,20 @@ public extension AudioProcessor {
...
+        
+        lastInputDevice = inputDeviceID
+    }
+    
+    func reStartRecordingLive(inputDeviceID: DeviceID? = nil, callback: (([Float]) -> Void)? = nil) throws {
+        try? setupAudioSessionForDevice()
+        
+        if inputDeviceID == lastInputDevice{
+            try audioEngine?.start()
+        }else{
```
Comment (ZachNagengast) [inline Sources/WhisperKit/Core/AudioProcessor.swift:607]:
```suggestion
        } else {
```

Comment (fengcunhan) [reply Sources/WhisperKit/Core/AudioProcessor.swift:607]: OK ，I will fix it

**Comment by fengcunhan** [comment]
> This is great, seems like there was the pause feature but was missing resume, suggested a couple changes for clarity.

Yes,i will use your suggestion.Thank you for review my code.

File: Sources/WhisperKit/Core/AudioProcessor.swift [hunk 6]
```diff
@@ -585,6 +593,22 @@ public extension AudioProcessor {
...
+        
+        if inputDeviceID == lastInputDevice{
+            try audioEngine?.start()
+        } else {
+            audioEngine = try setupEngine(inputDeviceID: inputDeviceID)
+        }
+        
+
         // Set the callback
         audioBufferCallback = callback
```
Comment (ZachNagengast) [inline Sources/WhisperKit/Core/AudioProcessor.swift:613]: Looking at it again, we probably won't want this to be overwritten when resuming if callback is nil and audioBufferCallback isn't nil

```suggestion
        // Set the callback only if the provided callback is not nil
        if let callback = callback {
            audioBufferCallback = callback
        }
```

**Review by ZachNagengast — APPROVED** [review]
Looks great! Thanks for contributing this @fengcunhan

**Comment by fengcunhan** [comment]
> Looks great! Thanks for contributing this @fengcunhan

You are welcome.Thanks for  your patient of review my begainer level swift code.This project is awesome,i like it

**Pull request merged**